frontend/server/src/main/java/com/amazonaws/ml/mms/http/ManagementRequestHandler.java

```diff
@@ -243,23 +243,30 @@ private void updateModelWorkers(
                 modelManager.updateModel(modelName, minWorkers, maxWorkers);
         if (!synchronous) {
             NettyUtils.sendJsonResponse(
-                    ctx, new StatusResponse("Worker updated"), HttpResponseStatus.ACCEPTED);
+                    ctx,
+                    new StatusResponse("Processing worker updates..."),
+                    HttpResponseStatus.ACCEPTED);
             return;
         }
         future.thenApply(
                         v -> {
+                            boolean status = modelManager.scaleRequestStatus(modelName);
+                            String response = "Workers scaled";
                             if (!v) {
+                                response = "Workers scaling in progress";
                                 if (onError != null) {
                                     onError.apply(null);
+                                    response = "Load failed... Deregistered model " + modelName;
                                 }
                                 NettyUtils.sendError(
                                         ctx,
                                         HttpResponseStatus.NOT_FOUND,
-                                        new ModelNotFoundException(
-                                                "Model not found: " + modelName));
+                                        new ModelNotFoundException(response));
                             } else {
-                                NettyUtils.sendJsonResponse(
-                                        ctx, new StatusResponse("Worker scaled"));
+                                if (!status) {
+                                    response = "Workers scaling in progress..";
+                                }
+                                NettyUtils.sendJsonResponse(ctx, new StatusResponse(response));
```

Comment:
We should send non 200 code back if status is false.

Reply:
OH. I thought it was a success since we already created the workers and we are just retrying to load again and again. What code would you suggest best reflects this? Should we send a 5XX error for sync-calls?

Reply:
How about 206?

Reply:
But 206 is partial content.. https://tools.ietf.org/html/rfc7233#section-4.1 .. RFC suggests that its for returning partial content... Quote from the RFC
```
   If a single part is being transferred, the server generating the 206
   response MUST generate a Content-Range header field, describing what
   range of the selected representation is enclosed, and a payload
   consisting of the range
```

Not sure if this would affect client behaviors...

Reply:
Agree, I think we can make up one: 210 partial success.

Reply:
Yup. Added that.